src/Microsoft.ML.Core/Utilities/Contracts.cs

```diff
@@ -584,6 +584,11 @@ public static void CheckParamValue<T>(this IExceptionContext ctx, bool f, T valu
             if (object.ReferenceEquals(val, null))
                 throw ExceptValue(ctx, paramName);
         }
+        public static void CheckStruct<T>(this IExceptionContext ctx, T val, string paramName) where T : struct
```

Comment:
>CheckStruct [](start = 27, length = 11)

I'm not sure I like this idea of a `CheckStruct`. I think rather I'd prefer that the checks on the columns not be this `CheckStruct`, but rather directly against where it `IsValid`.

Reply:
Just change `CheckStruct(col, nameof(col))` to `Contracts.Check(col.IsValid, nameof(col))`
`

---
In reply to: [238850745](https://github.com/dotnet/machinelearning/pull/1822#discussion_r238850745) [](ancestors = 238850745)